### PR TITLE
ss-258: Connect budgets to backend

### PR DESCRIPTION
**Pull request by teslmik**
chrome_SWARqGXVel

### CODE REVIEW

**Review by anton-otroshchenko — APPROVED** [review]

File: frontend/src/bundles/common/pages/budgets/budget-details/components/budget-modal/budget-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+import {
+    RenderCurrency,
+    RenderDate,
+    RenderMultiDropdown,
+    RenderRecurrence,
+} from '../components';
+import styles from './styles.module.scss';
+
+interface BudgetModalProperties {
```
Comment (Evcat6) [inline frontend/src/bundles/common/pages/budgets/budget-details/components/budget-modal/budget-modal.tsx:23]: We use types in components props typization and i recomend to use just just "Properties".

File: frontend/src/bundles/common/pages/budgets/budget-details/helpers/calculate-budget-details.helper.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            break;
+        }
+    }
+    const lastDate = new Date(endDate).toLocaleDateString('en-US', {
+        month: 'short',
+        day: '2-digit',
+        year: 'numeric',
+    });
+    const timeRemainingMs = endDate.getTime() - Date.now();
+    const daysRemaining = Math.ceil(timeRemainingMs / (1000 * 60 * 60 * 24));
```
Comment (Evcat6) [inline frontend/src/bundles/common/pages/budgets/budget-details/helpers/calculate-budget-details.helper.ts:54]: Magic numbers

File: frontend/src/bundles/common/pages/budgets/budgets.tsx [hunk 3]
```diff
@@ -75,14 +72,13 @@ const Budgets: React.FC<Properties> = ({ budgetCards }) => {
...
-                onSubmit={handleClickClose}
-                submitButtonName={'Create budget'}
-                Header={'Add new budget'}
-                Body={'New budget body...'}
-            />
+            <div className={styles.modal}>
+                <BudgetModal
+                    isShown={active}
+                    onClose={handleCancel}
+                    // budget={budget}
```
Comment (Evcat6) [inline frontend/src/bundles/common/pages/budgets/budgets.tsx:79]: We need this comment?

**Review by Evcat6 — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/pages/budgets/budget-details/helpers/calculate-budget-details.helper.ts (hunk 2 above)
Comment (teslmik) [reply frontend/src/bundles/common/pages/budgets/budget-details/helpers/calculate-budget-details.helper.ts:54]: this is the code from another task

File: frontend/src/bundles/common/pages/budgets/budget-details/components/budget-modal/budget-modal.tsx (hunk 1 above)
Comment (teslmik) [reply frontend/src/bundles/common/pages/budgets/budget-details/components/budget-modal/budget-modal.tsx:23]: this is the code from another branch

File: frontend/src/bundles/common/pages/budgets/budgets.tsx (hunk 3 above)
Comment (teslmik) [reply frontend/src/bundles/common/pages/budgets/budgets.tsx]: Fixed

**Review by anton-otroshchenko — APPROVED** [review]

**Review by Holovashcenko — DISMISSED** [review]

**Comment by nikita-remeslov** [comment]
Conflicts

File: frontend/src/bundles/common/pages/budgets/budget-details/components/budget-modal/budget-modal.tsx [hunk 4]
```diff
@@ -18,59 +21,97 @@ import {
...
+        Boolean(watch('amount')),
+        Boolean(watch('currency')),
+        Boolean(watch('recurrence')),
+        Boolean(watch('categories')),
+        Boolean(watch('startDate')),
+    ];
+
+    const categoriesId = budget?.categories.map((it) => it.id);
+    const watchCategoryId = (watch('categories') as unknown as categoryType[])
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/pages/budgets/budget-details/components/budget-modal/budget-modal.tsx:66]: ?

File: frontend/src/bundles/common/pages/budgets/budget-details/components/budget-modal/budget-modal.tsx [hunk 5]
```diff
@@ -18,59 +21,97 @@ import {
...
+        ?.map((it) => it.id);
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/pages/budgets/budget-details/components/budget-modal/budget-modal.tsx:68]: ?

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**